src/lowering/desugar.ml

```diff
@@ -216,7 +216,16 @@ and exp' at note = function
   | S.OldE e -> (oldE (exp e)).it
   | S.IfE (e1, e2, e3) -> I.IfE (exp e1, exp e2, exp e3)
   | S.SwitchE (e1, cs) -> I.SwitchE (exp e1, cases cs)
-  | S.TryE (e1, cs) -> I.TryE (exp e1, cases cs)
+  | S.TryE (e1, cs, None) -> I.TryE (exp e1, cases cs, None)
```

Comment:
Hmm, doing the transformation half in desugaring and half later seems a bit dangerous to me. Why not simplify desugaring to just name the thunk and do the rest later...

But I see that is on your TODO list already...

Reply:
This is fixed now, the desugaring step only creates the thunk, which is then lowered in `await.ml`. The interpreters are now functionally identical too :-)